runtime/compilers/rillv1/parse_api.go

```diff
@@ -0,0 +1,53 @@
+package rillv1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+)
+
+// APIYAML
+type APIYAML struct {
+	SQL        string         `yaml:"sql"`
+	MetricsSQL *MetricSQLYAML `yaml:"metrics_sql"`
```

Comment:
1. Would suggest changing `metrics_sql` to `metrics` (since `SQL` is a property of it)
2. Since the `MetricsSQLYAML` type isn't used, let's use an anonymous struct, similar to the other YAML types in this package (e.g. see `ReportYAML`).

Reply:
Understood. Also considering other review comments I believe even `metrics` isn't required and we can directly map `resolver_properties` to `map[string]any` and pass to `APIResolverOptions` ?